rest-lib/src/main/scala/com/gu/mediaservice/lib/auth/Authorisation.scala

```diff
@@ -22,13 +24,36 @@ class Authorisation(provider: AuthorisationProvider, executionContext: Execution
   def actionFilterFor(permission: SimplePermission): ActionFilter[Request] =
     actionFilterFor(
       permission,
-      respondError(Unauthorized, "permission-denied", s"You do not have permission to ${permission.name}")
+      unauthorized(permission)
     )
 
   object CommonActionFilters {
     lazy val authorisedForUpload = actionFilterFor(UploadImages)
   }
 
+  private def isUploaderOrHasPermission(
+                                         principal: Principal,
+                                         uploadedBy: String,
+                                         permission: SimplePermission
+                                       ) = {
+    principal match {
+      case user: UserPrincipal =>
+        if (user.email.toLowerCase == uploadedBy) {
+          true
+        } else {
+          hasPermissionTo(permission)(principal)
+        }
+      case MachinePrincipal(ApiAccessor(_, Internal), _) => true
+      case _ => false
+    }
+  }
+
+  def canUserWriteMetadata(principal: Principal, uploadedBy: String): Boolean =
+    isUploaderOrHasPermission(principal, uploadedBy, EditMetadata)
+
+  def canUserDeleteImage(principal: Principal, uploadedBy: String): Boolean =
+    isUploaderOrHasPermission(principal, uploadedBy, DeleteImage)
```

Comment:
Super happy for `isUploaderOrHasPermission` to be here but I'm quite keen to avoid embedding anything specific to a particular permission in this class. Can these wrapper methods be colocated in the places they are used?

Reply:
Yes, I moved them here because canUserWriteMetadata is used by both media-api and metadata-editor services, but I can move the wrappers back to media-api and use isUploaderOrHasPermission directly in metadata-editor (or maybe move them to a common class? not sure where...)

Reply:
Fixed